xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -277,6 +289,12 @@ bool CRenderer::HasOverlay(int idx)
 
 void CRenderer::SetVideoRect(CRect &source, CRect &dest, CRect &view)
 {
+  if (m_rv != view)
```

Comment:
Glad to see this worked. Without wanting to be a pain in the ass do you mind turning this into a some sort of "callback" function call? Something on the lines of:

```
if (m_rv != view)
  OnViewChanged()
```

Also I am not sure if you shouldn't simply give up if the lock is locked - depends on who calls `SetVideoRect`. I am assuming it is the render thread. If that the case locking is something that you should try to avoid (or employ some sort of lazy approach: try lock -> change, islocked -> give up. The event (`m_isSettingsChanged`) will be recorded anyway

Reply:
I'd have to actually tested. I've done something similar recently (didn't went in but the implementation might be similar):

https://github.com/xbmc/xbmc/pull/21111/files#diff-204721060df48a53f6d011516f7680586a061ebe67336ceeecc1cce296c2a08aR439-R447